File: frontend/src/services/prescriptionService.js
import apiClient from './apiClient';

const prescriptionService = {
  async create(payload) {
    const { data } = await apiClient.post('/prescriptions', payload);
    return data.data;
  },
  async getById(id) {
    const { data } = await apiClient.get(`/prescriptions/${id}`);
    return data.data;
  },
  async getByAppointment(appointmentId) {
    const { data } = await apiClient.get(`/prescriptions/appointment/${appointmentId}`);
    return data.data;
  },
  async getByPatient(patientId, params = {}) {
    const { data } = await apiClient.get(`/prescriptions/patient/${patientId}`, { params });
    return data.data;
  },
  async getByDoctor(doctorId, params = {}) {
    const { data } = await apiClient.get(`/prescriptions/doctor/${doctorId}`, { params });
    return data.data;
  },
};

export default prescriptionService;

File: backend/src/main/java/com/hospital/hms/service/PrescriptionService.java
package com.hospital.hms.service;

import com.hospital.hms.dto.request.PrescriptionRequest;
import com.hospital.hms.dto.response.PrescriptionResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface PrescriptionService {

    PrescriptionResponse create(PrescriptionRequest request, Long doctorUserId);

    PrescriptionResponse getById(Long id);

    PrescriptionResponse getByAppointment(Long appointmentId);

    Page<PrescriptionResponse> getByPatient(Long patientId, Pageable pageable);

    Page<PrescriptionResponse> getByDoctor(Long doctorId, Pageable pageable);

    /** Attaches an AI-generated plain-language explanation; called from the AI module. */
    PrescriptionResponse attachAiSummary(Long id, String aiSummary);
}
